src/CloudFlare/IpRewrite.php

```diff
@@ -2,65 +2,72 @@
 
 namespace CloudFlare;
 
-class IpRewrite {
-    static protected $is_cf = NULL;
-    static protected $original_ip = FALSE;
-    static protected $rewritten_ip = FALSE;
-    
-    static protected $cf_ipv4 = array(
-        "199.27.128.0/21",
-        "173.245.48.0/20",
-        "103.21.244.0/22",
-        "103.22.200.0/22",
-        "103.31.4.0/22",
-        "141.101.64.0/18",
-        "108.162.192.0/18",
-        "190.93.240.0/20",
-        "188.114.96.0/20",
-        "197.234.240.0/22",
-        "198.41.128.0/17",
-        "162.158.0.0/15",
-        "104.16.0.0/12",
-        "172.64.0.0/13"
+require_once 'vendor/autoload.php';
+
+class IpRewrite
+{
+    protected static $is_cf = null;
+    protected static $original_ip = false;
+    protected static $rewritten_ip = false;
+
+    protected static $cf_ipv4 = array(
+        '199.27.128.0/21',
+        '173.245.48.0/20',
+        '103.21.244.0/22',
+        '103.22.200.0/22',
+        '103.31.4.0/22',
+        '141.101.64.0/18',
+        '108.162.192.0/18',
+        '190.93.240.0/20',
+        '188.114.96.0/20',
+        '197.234.240.0/22',
+        '198.41.128.0/17',
+        '162.158.0.0/15',
+        '104.16.0.0/12',
+        '172.64.0.0/13',
+        '131.0.72.0/22',
     );
-    
-    static protected $cf_ipv6 = array(
-        "2400:cb00::/32",
-        "2606:4700::/32",
-        "2803:f800::/32",
-        "2405:b500::/32",
-        "2405:8100::/32"
+
+    protected static $cf_ipv6 = array(
+        '2400:cb00::/32',
+        '2606:4700::/32',
+        '2803:f800::/32',
+        '2405:b500::/32',
+        '2405:8100::/32',
     );
-    
+
     // Returns boolean
     public static function isCloudFlare()
     {
         self::rewrite();
+
         return self::$is_cf;
```

Comment:
You're correct it is a bit confusing. I did not touch that part but it works

```
protected static function rewrite()
    {
        // only should be run once per page load
        if (self::$is_cf !== null) {
            return;
        }

        // Set this based on the presence of the header, so it is true even if IP has already been rewritten
        self::$is_cf = isset($_SERVER['HTTP_CF_CONNECTING_IP']) ? true : false;
```

So basically it'll $is_cf will never return null